tools/calyx-writer/lib.rs

```diff
@@ -857,22 +874,21 @@ impl CalyxWriter for Component {
 pub struct Program {
     imports: Vec<Import>,
     /// inv: no element is removed from this array
-    comps: Vec<Rc<RefCell<Component>>>,
+    comps: Vec<RRC<Component>>,
 }
 
 impl Program {
     /// Constructs an empty program.
     pub fn new() -> Self {
-        Self {
-            imports: vec![],
-            comps: vec![],
-        }
+        Self::default()
     }
 
     /// Imports the calyx standard library file `path`.
-    pub fn import<S: ToString>(&mut self, path: S) {
+    ///
+    /// Requires: `path` is a well-formed path.
+    pub fn import<S: AsRef<str>>(&mut self, path: S) {
         self.imports.push(Import {
-            path: path.to_string(),
+            path: PathBuf::from_str(path.as_ref()).expect("precondition"),
```

Comment:
I would make this error message marginally more helpful. Something like "malformed import path". In future versions of the library you could escalate to a proper error type and result but that's not needed here